se/se_epub.py

```diff
@@ -79,9 +79,6 @@ class SeEpub:
 	An SE epub can have various operations performed on it, including recomposing and linting.
 	"""
 
-	path = ""
-	metadata_file_path = None
-	metadata_xhtml = None
```

Comment:
These are used in several places, why are they removed?

Reply:
I made the change because mypy detected that the types of the class variables don't agree with the types of the corresponding instance variables when they are initialized in `__init__()`.

The only purpose those class variables serve is to provide default values for the instance variables with the same name if they are never initialized. Since all three instance variables are initialized in the `__init__()` method I didn't see a reason to keep them around.

Another solution to the type problem would be to change the class variables, like so:

	path = Path()
	metadata_file_path = Path()
	metadata_xhtml = ""

But those class variables would never be used unless the implementation of `__init__` changed.

Reply:
OK, I see. I'd like to keep the definitions there so somebody reading the class can see that those properties are public parts of the class and available for use. Can we add type hints to keep mypy happy, for example `metadata_file_path: Path = None`?

Reply:
If not then I think just declaring them as empty `Path()`s is fine. They were probably initialized to none/empty string because before `Path` was available, paths were passed around the code as strings.

Reply:
The problem with assigning `None` is that you then need to change the type from `Path` to `Optional[Path]` which causes all sorts of complaints from mypy. And `Optional[Path]` isn't really accurate anyways because you always set the value in the constructor.

I understand the desire to document the public fields, so I think the best bet in the short term is to just add the unused Path() objects. I will update the code.

In the longer term, you might want to consider adding a class docstring that lists the public instance variables instead. The class variables aren't really intended for documenting instance variables and most Python code isn't written that way, so it makes a bit confusing for someone new to this code base. (Or at least it was confusing for me.)

Reply:
OK great. We can look in to converting to docstrings a little later. This toolset is my first Python project so much of it is learning this best practice stuff as I go along.